unit_tests/util/json_render.cpp

```diff
@@ -27,4 +27,11 @@ BOOST_AUTO_TEST_CASE(integer)
     BOOST_CHECK_EQUAL(str, "42");
 }
 
+BOOST_AUTO_TEST_CASE(test_json_issue_6531) {
```

Comment:
@Rejudge-F  Cool that you found this `g` suffix! Thanks! May I ask you to add 3-4 test cases more in order to make sure that the rest of the things work as expected? e.g. `42.1`, `42.0`, `0.12345678912345`? 

Let me know if you don't have time on it and I will try to do that on my own.

Reply:
I will do it.